### PR TITLE
Add tsconfig to non monorepo plugin.

### PR DESCRIPTION
**Pull request by feiming**
## Add tsconfig to non monorepo plugins. 

Skip adding tsconfig in monorepo plugins scaffolding
Skip tsconfig check in plugin diff
Add template tsconfig for non monorepo plugins

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/spotify/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

### CODE REVIEW

File: packages/cli/templates/default-plugin/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
+  "extends": "@backstage/cli/config/tsconfig.json",
+  "include": [
+    "src",
+    "dev",
+    "migrations"
```
Comment (Rugvip) [inline packages/cli/templates/default-plugin/tsconfig.json:6]: No need to have this for frontend plugins

**Review by Rugvip — COMMENTED** [review]
Yep 👍 , just a minor nit

**Comment by benjdlambert** [comment]
Maybe you could duplicate this into the `default-plugin-backend` skeleton too? It makes sense the put the `migrations` key in that `tsconfig` instead

File: packages/cli/templates/default-plugin/tsconfig.json (hunk 1 above)
Comment (feiming) [reply packages/cli/templates/default-plugin/tsconfig.json]: done

**Review by benjdlambert — APPROVED** [review]

**Pull request merged**